packages/core/state/selection/reducer.ts

```diff
@@ -169,14 +171,33 @@ export default makeReducer<SelectionStateBranch>(
             ...state,
             fileSelection: action.payload,
         }),
-        [SET_ANNOTATION_HIERARCHY]: (state, action) => ({
-            ...state,
-            annotationHierarchy: action.payload,
-            availableAnnotationsForHierarchyLoading: true,
+        [SET_ANNOTATION_HIERARCHY]: (state, action) => {
+            // combine recent hierarchy annotations with payload
+            const recentHierarchyAnnotationsUnfiltered = [
+                ...castArray(action.payload),
+                ...state.recentHierarchyAnnotations,
+            ];
 
-            // Reset file selections when annotation hierarchy changes
-            fileSelection: new FileSelection(),
-        }),
+            // remove duplicates and limit to 7 most recent
+            const recentHierarchyAnnotations = [
+                ...new Map(
+                    recentHierarchyAnnotationsUnfiltered.map((annotation) => [
+                        annotation.name,
+                        annotation,
+                    ])
+                ).values(),
+            ].slice(0, 7);
```

Comment:
```suggestion
            const recentHierarchyAnnotations = lodash.uniqBy(recentHierarchyAnnotationsUnfiltered, 'name').slice(0, 7);
```
I think this can be simplified with a change like this.